### PR TITLE
tord dijkstras fp

### PR DESCRIPTION
**Pull request by Kazads**

### CODE REVIEW

File: FPDijkstrasTord/src/js/Dijkstras.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    var unvisitedNodes = graph.slice();
+    var currentNode = 0;
+    var currentValue = Number.MAX_SAFE_INTEGER;
+    var pathAndDist = unvisitedNotEmpty(graph,edges,startNode,endNode,path,dist,unvisitedNodes,currentNode,currentValue);
+    return pathAndDist[1][graph.indexOf(endNode)];
+  }else{
+    return 0;
+  }
+}
+
```
Comment (KimSvenSand) [inline FPDijkstrasTord/src/js/Dijkstras.js:21]: Copyyyyy cats :octocat:

**Review by KimSvenSand — CHANGES_REQUESTED** [review]
Be careful with mutating data. Okay to mutate local data in functions. Some more structure and better naming  of functions. Try to use filter and map if possible. Use input -> output thinking.

File: FPDijkstrasTord/src/js/Dijkstras.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+export function createPathAndDist(length,path,dist){
+  if(length > 1){
```
Comment (emilfolino) [inline FPDijkstrasTord/src/js/Dijkstras.js:23]: Muterar path and dist.

File: FPDijkstrasTord/src/js/Dijkstras.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    dist = arr[1];
+  }
+  path.push("");
+  dist.push(Number.MAX_SAFE_INTEGER);
+  return [path,dist];
+
+}
+
+export function unvisitedNotEmpty(graph, edges,startNode,endNode,path,dist,unvisitedNodes,currentNode,currentValue){
+  if(unvisitedNodes.length > 0 && unvisitedNodes.indexOf(endNode) != -1){
```
Comment (emilfolino) [inline FPDijkstrasTord/src/js/Dijkstras.js:35]: muterar path och dist

File: FPDijkstrasTord/src/js/Dijkstras.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    var arr = unvisitedNotEmpty(graph,edges,startNode,endNode,path,dist,unvisitedNodes,currentNode,currentValue);
+    path = arr[0];
+    dist = arr[1];
+  }
+  return [path,dist];
+
+}
+
+export function findCurrentNode(graph,graphLength,unvisitedNodes,dist,currentNode,currentValue){
+  if(graphLength > 0){
```
Comment (emilfolino) [inline FPDijkstrasTord/src/js/Dijkstras.js:52]: muterar currentNode och currentValue

File: FPDijkstrasTord/src/js/Dijkstras.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+  if(dist[graphLength] < currentValue && unvisitedNodes.indexOf(graph[graphLength]) != -1){
+    currentNode = graphLength;
+    currentValue = dist[graphLength];
+  }
+  return [currentNode,currentValue];
+}
+
+export function assignPathAndDist(graph,edges,edgeLength,path,dist,currentNode){
+  if(edgeLength > 0){
```
Comment (emilfolino) [inline FPDijkstrasTord/src/js/Dijkstras.js:66]: Muterar path och dist.

File: FPDijkstrasTord/src/js/Dijkstras.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  }else if(currentToNode == graph[currentNode] && dist[graph.indexOf(currentFromNode)] > dist[currentNode] + parseInt(currentEdgeLength)){
+    dist[graph.indexOf(currentFromNode)] = parseInt(dist[currentNode]) + parseInt(currentEdgeLength);
+    path[graph.indexOf(currentToNode)] = path[currentNode] +","+ currentToNode+"-"+currentFromNode+"="+currentEdgeLength;
+  }
+
+  return [path,dist];
+}
+
+export function removeVisitedItem(graph,unvisitedNodes,unvisitedLength,currentNode,newArray){
+  if(unvisitedLength > 0){
```
Comment (emilfolino) [inline FPDijkstrasTord/src/js/Dijkstras.js:87]: Muterar newArray.

**Review by emilfolino — COMMENTED** [review]
Undvik mutation, skapa kopior.

**Pull request closed without merging**